src/main/scala/org/broadinstitute/dsde/workbench/sam/api/StandardUserInfoDirectives.scala

```diff
@@ -126,17 +113,19 @@ object StandardUserInfoDirectives {
     }
   }
 
-  private def lookUpByGoogleSubjectId(googleSubjectId: GoogleSubjectId, directoryDAO: DirectoryDAO, samRequestContext: SamRequestContext): IO[WorkbenchUserId] =
+  private def lookUpByGoogleSubjectId(googleSubjectId: GoogleSubjectId, directoryDAO: DirectoryDAO, samRequestContext: SamRequestContext): IO[SamUser] =
     for {
-      subject <- directoryDAO.loadSubjectFromGoogleSubjectId(googleSubjectId, samRequestContext)
-      userInfo <- subject match {
-        case Some(uid: WorkbenchUserId) => IO.pure(uid)
-        case Some(_) =>
-          IO.raiseError(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Conflict, s"subjectId $googleSubjectId is not a WorkbenchUser")))
-        case None =>
-          IO.raiseError(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Forbidden, s"Google Id $googleSubjectId not found in sam")))
+      maybeSamUser <- directoryDAO.loadUserByGoogleSubjectId(googleSubjectId, samRequestContext)
+      samUser <- maybeSamUser match {
+        case Some(samUser) => IO.pure(samUser)
+        case None => directoryDAO.loadSubjectFromGoogleSubjectId(googleSubjectId, samRequestContext).flatMap {
+          case Some(_) =>
+            IO.raiseError(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Conflict, s"subjectId $googleSubjectId is not a WorkbenchUser")))
+          case None =>
+            IO.raiseError(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Forbidden, s"Google Id $googleSubjectId not found in sam")))
+        }
```

Comment:
The old block of code called loadSubjectFromGoogleSubjectId, which queries both pet and user tables and could return either a pet id or user id, then throws an exception in all cases other than user id.

The new block of code looks up a user (because we want the full user object) and if it doesn't find one does a follow up query to see if there is a pet before decide which exception to throw.

Reply:
actually, the follow up query is unnecessary because the calling code already handles pets... I will remove it in the next iteration

Reply:
+1, seems the pet case is already handled before this… so maybe we don't need the `loadSubjectFromGoogleSubjectId` query and can throw a generic error if the user is not found.